File: optimise/v01.c
#include <stdio.h>
#include <numeric>


int main(){
    int N, combinations;
    while (scanf("%d", &N) && N){
        combinations = 0;
        for(int x=1; x < N; x++){
            for(int y=1; y < N; y++){
                for(int z=1; z < N; z++){
                    if(std::gcd(std::gcd(x, y), z) == 1 &&
                       x * x + y * y == z * z &&
                       x < y && y < z){
                            combinations += 1;
                    }
                }
            }
        }
        printf("%d\n", combinations);
    }
    return 0;
}

File: optimise/v01.py
import sys
from math import gcd


for line in sys.stdin:
    N = int(line)
    combinations = 0
    for x in range(N + 1):
        for y in range(N + 1):
            for z in range(N + 1):
                if gcd(gcd(x, y), z) == 1 and\
                   x ** 2 + y ** 2 == z ** 2 and x < y < z:
                    combinations += 1
    print(combinations)
